### ISSUE TITLE
Prototypes are computed in train mode

### ISSUE THREAD

**Issue by PeterTor** [sicara/easy-few-shot-learning#18]
hello i have a question. 
Is it correct that the backbone should adjust the batch norm statistics when processing the prototypes (s. https://arxiv.org/abs/1603.04779)? I get significant differences when I set the backbone to eval() in the method process_support_set() before computing the prototytpes like:

    def process_support_set(
        self,
        support_images: torch.Tensor,
        support_labels: torch.Tensor,
    ):

        self.backbone.eval() # <- Ive added this line the rest is unchanged
        support_features = self.backbone.forward(support_images)
        self.prototypes = compute_prototypes(support_features, support_labels)

This way the backbone wont update the BN statistics. However I dont know if this is intended and some clarification would be really helpful.

Thanks in advance

**Comment by ebennequin** [comment]
Hi,

My understanding is that we want to be in train mode when training your prototypical networks, and in eval mode when evaluating your prototypical networks, regardless of whether we are processing the support set or the query set.
If you put an `eval()` in `process_support_set()`, then the training behavior of `AbstractMetaLearner` models will be affected. For instance, in the `fit()` method, the model will never actually be in train mode, since you switch it to eval before the first forward pass.

Do you think it should behave differently?
